sinks/cache/cache_impl.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"time"
+
+	source_api "github.com/GoogleCloudPlatform/heapster/sources/api"
+	cadvisor_api "github.com/google/cadvisor/info/v1"
+)
+
+// TODO: Add unit tests
+type realCache struct {
+	bufferDuration time.Duration
+	// Map of pod UIDs to pod cache entry.
+	Pods map[string]*PodElem
+	// Map of node hostnames to node cache entry.
+	Nodes map[string]*NodeElem
+}
+
+const gcDuration = time.Second
+
+func (rc *realCache) newContainerElem() *ContainerElem {
+	return &ContainerElem{
+		ContainerSpec:  NewGCStore(NewTimeStore(), rc.bufferDuration, gcDuration),
```

Comment:
GCStore starts a thread for garbage collection, this would be a thread per container we are storing which could get quite huge. WDYT if we enforce the eviction age on the TimedStore in an insert and we then GC ContainerElements?